### PR TITLE
Replace Compiler::stripWhitespace with native implementation

### PR DESCRIPTION
**Pull request by Chi-teck**
This reduces the size of composer.phar file about 2% and makes compiler a bit more simple.

### CODE REVIEW

File: src/Composer/Compiler.php [hunk 1]
```diff
@@ -215,40 +214,6 @@ private function addComposerBin($phar)
         $phar->addFromString('bin/composer', $content);
     }
 
-    /**
-     * Removes whitespace from a PHP source string while preserving line numbers.
```
Comment (Chi-teck) [inline src/Composer/Compiler.php:219]: I do not know what was the reason to preserve line numbers. With this PR newlines will be removed.

**Comment by Seldaek** [comment]
As you pointed out, this removes newlines, which is unacceptable to us as it means the bug reports we get with stack traces are impossible to match to real source. We'd need to store sourcemaps which would be even more complex than keeping the newlines.

File: src/Composer/Compiler.php (hunk 1 above)
Comment (stof) [reply src/Composer/Compiler.php:219]: the reason is that 100% of bug reports we receive are providing stack traces generated by the phar. Preserving line numbers makes them actionable.

Comment (Chi-teck) [reply src/Composer/Compiler.php:219]: We could leave a note about that reason in the compiler.

**Pull request closed without merging**